File: exercism-solutions/clojure/anagram/src/anagram.clj
(ns anagram)

(defn- anagram? [word candidate]
  (let [candidate (clojure.string/lower-case candidate)]
    (and
     (not= word candidate)
     (= (sort word) (sort candidate)))))

(defn anagrams-for [word prospect-list]
  (let [word (clojure.string/lower-case word)]
    (filter #(anagram? word %) prospect-list)))

File: exercism-solutions/julia/anagram/anagram.jl

function detect_anagrams(subject, candidates)
    sorted(s) = sort(collect(lowercase(s)))

    subject = lowercase(subject)
    subject_sorted = sorted(subject)

    function check(candidate)
        candidate = lowercase(candidate)
        (candidate != subject) && (sorted(candidate) == subject_sorted)
    end

    filter(check, candidates)
end
